examples/common/instance_utils.py

```diff
@@ -19,7 +19,18 @@ def open_level(instance, scene_id, map_id=""):
     spear.log("level_name: ", level_name)
 
     instance.engine_service.begin_tick()
-    instance.game_world_service.open_level(level_name)
+    gameplay_statics_class = instance.game_world_service.get_static_class(class_name="UGameplayStatics")
+    assert gameplay_statics_class
+    gameplay_statics_default_object = instance.game_world_service.get_default_object(uclass=gameplay_statics_class, create_if_needed=False)
+    assert gameplay_statics_default_object
+    open_level_func = instance.game_world_service.find_function_by_name(uclass=gameplay_statics_class, name="OpenLevel", include_super_flag="IncludeSuper")
+    assert open_level_func
+    instance.engine_service.tick()
+    instance.engine_service.end_tick()
+
+    instance.engine_service.begin_tick()
+    instance.game_world_service.call_function(uobject=gameplay_statics_default_object, ufunction=open_level_func, args={"WorldContextObject": gameplay_statics_default_object, "LevelName": level_name})
```

Comment:
Why does this function call happen in a separate Unreal frame? Couldn't it happen in the previous frame immediately after `find_function_by_name`? Also, these `WorldContextObject` arguments are handled automatically by Unreal's Blueprint system. We should provide comparable behavior. See this link and search for `WorldContext` if you're curious to find out more,

https://docs.unrealengine.com/4.27/en-US/ProgrammingAndScripting/GameplayArchitecture/Metadata

So, let's change our code as follows,

```
instance.game_world_service.call_function(uobject=gameplay_statics_default_object, ufunction=open_level_func, args={"LevelName": level_name})
```

For this to work, we will need to make the following changes throughout the call stack,

```
def call_function(self, uobject, ufunction, args={}, world_context="WorldContextObject"):
    arg_strings = { arg_name: json.dumps(arg) for arg_name, arg in args.items() }
    return self._rpc_client.call("game_world_service.call_function", uobject, ufunction, arg_strings, world_context)


unreal_entry_point_binder->bindFuncUnreal("game_world_service", "call_function",
    [this](
        const uint64_t& uobject,
        const uint64_t& ufunction,
        const std::map<std::string, std::string>& args,
        const std::string& world_context) -> std::map<std::string, std::string> {

        return Unreal::callFunction(world_, reinterpret_cast<UObject*>(uobject), reinterpret_cast<UFunction*>(ufunction), args, world_context);
    });


// .h
std::map<std::string, std::string> Unreal::callFunction(UWorld* world, UObject* uobject, UFunction* ufunction, const std::map<std::string, std::string>& args, const std::string& world_context = "WorldContextObject")


// .cpp
std::map<std::string, std::string> Unreal::callFunction(UWorld* world, UObject* uobject, UFunction* ufunction, const std::map<std::string, std::string>& args, const std::string& world_context)
{
    SP_ASSERT(world);
    SP_ASSERT(uobject);
    SP_ASSERT(ufunction);

    // ...

    // Set property values.
    for (auto& property_desc : property_descs) {
        std::string property_name = toStdString(property_desc.property_->GetName());

        // If the caller has flagged the property name as the "world context", then set the property using
        // the input world pointer, and make sure the name is not present in args. Strictly speaking, we
        // could assign the world pointer directly to *property_desc.value_ptr, but this could lead to
        // undefined behavior if the property is not actually a pointer. So we assign via setPropertValueFromString(...)
        // because this will avoid undefined behavior.
        if (property_name == world_context) {
            SP_ASSERT(!Std::containsKey(args, property_name));
            setPropertValueFromString(property_desc, Std::toStringFromPtr(world));
        }

        // If the property name is in args, then set it using the string in args, and make make sure the name
        // has not also been flagged as the "world context".
        if (Std::containsKey(args, property_name)) {
            SP_ASSERT(property_name != world_context);
            setPropertyValueFromString(property_desc, args.at(property_name));
        }
    }

    // ...
}
```

Reply:
Wasn't aware that Blueprint function calls automatically handled this `WorldContextParameter` argument for us. Thanks for pointing it out!